src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -145,6 +149,7 @@
   [config :- {schema/Keyword schema/Any}]
   (-> config
       (update-in [:compile-mode] #(keyword (or % default-jruby-compile-mode)))
+      (update-in [:compat-version] #(str (or % default-jruby-compat-version)))
```

Comment:
I played around with defining `compat-version: 2.0` in a HOCON config.  I found that it converted the value to just an integer of '2' in memory.  If we're doing the `str` here just as a convenience to allow callers to pass the original value from the config on through to jruby-utils without touching it and we want to, for convenience, allow for users to be able to put the value in the config with or without quotes, we may want to put a little extra code in here to allow for '2' to be treated as '2.0'.  Here's a hacky change I tried that seemed to work for that here:

```clj
(update-in [:compat-version]
  #(if % (str (if (integer? %) (double %) %)) default-jruby-compat-version))
```